presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -601,9 +777,39 @@ public Void visitSpecialForm(SpecialFormExpression specialForm, Context context)
                 return null;
             }
 
+            /**
+             *
+             * Adds lambda subfields from the context to the list of the required subfields of the field/subscript/subfield provided in parameter 'input'. This function should be
+             * invoked
+             * once we reached leaf node while visiting the expression tree. Effectively, it prunes all unaccessed subfields of the 'input'.
+             *
+             * @param context - SubfieldExtractor context
+             * @param input - input field, subscript, or subfield, for which lambda subfields were collected.
+             */
+            @Nullable
+            private void addRequiredLambdaSubfields(Context context, Subfield input)
+            {
+                if (!isPushDownSubfieldsFromLambdasEnabled || !context.getLambdaSubfields().isPresent()) {
+                    return;
+                }
+                Set<Subfield> lambdaSubfields = context.getLambdaSubfields().get();
+                for (Subfield lambdaSubfield : lambdaSubfields) {
+                    List<PathElement> newPath = ImmutableList.<PathElement>builder()
+                            .addAll(input.getPath())
+                            .addAll(lambdaSubfield.getPath())
+                            .build();
+                    context.subfields.add(new Subfield(input.getRootName(), newPath));
+                }
+                context.setLambdaSubfields(Optional.empty());
```

Comment:
it seems not quite right to me from a division of responsiblitiy perpective to set lambda subfields to empty here.  Setting this field to empty is more about how the context will be used next than the work this function is doing, which is something the caller of this function should know about, but this function shouldn't care about at all.